### PR TITLE
openapi: fix NPE when importing with body with empty dictionary

### PR DESCRIPTION
**Pull request by FiveOFive**
Fix zaproxy/zaproxy#7808

### CODE REVIEW

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java [hunk 1]
```diff
@@ -1197,6 +1197,25 @@ void shouldUseTargetUrlForOperationServers() throws Exception {
                         "PATCH http://server8.localhost/v2/operations/with/servers"));
     }
 
+    @Test
+    void shouldConvertEmptyContentDictionary() {
+        String definition = "openapi: 3.0.0\n"
```
Comment (kingthorin) [inline addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java:1202]:
```suggestion
        // Given
        String definition = "openapi: 3.0.0\n"
```

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java [hunk 2]
```diff
@@ -1197,6 +1197,25 @@ void shouldUseTargetUrlForOperationServers() throws Exception {
...
+        + "  /api/test:\n"
+        + "    post:\n"
+        + "      requestBody:\n"
+        + "        content: {}\n"
+        + "      responses:\n"
+        + "        200:\n"
+        + "          description: OK\n";
+
+        final SwaggerConverter converter = new SwaggerConverter(WELLFORMED_URL, null, definition, null);
+        assertDoesNotThrow(() -> converter.getRequestModels());
```
Comment (kingthorin) [inline addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java:1216]:
```suggestion
        // When / Then
        assertDoesNotThrow(() -> converter.getRequestModels());
```

**Comment by kingthorin** [comment]
Looks like the code needs a spotlessApply

**Comment by kingthorin** [comment]
CHANGELOG.md should also be updated.

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java [hunk 3]
```diff
@@ -1197,6 +1197,29 @@ void shouldUseTargetUrlForOperationServers() throws Exception {
                         "PATCH http://server8.localhost/v2/operations/with/servers"));
     }
 
+    @Test
+    void shouldConvertEmptyContentDictionary() {
```
Comment (thc202) [inline addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java:1201]: This really should be in a `RequestModelConverterUnitTest` not this class.

File: addOns/openapi/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Changed
+- Fix null pointer exception for OpenAPI import with empty dictionary (Issue 7808)
```
Comment (thc202) [inline addOns/openapi/CHANGELOG.md:8]: - This should be under `### Fixed`.
- End the sentence with a dot.
- `for OpenAPI import` → `when importing a definition`, this add-on applies only to OpenAPI.

Comment (thc202) [reply addOns/openapi/CHANGELOG.md]: The dot could still be added.

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/RequestModelConverterUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
```
Comment (thc202) [inline addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/RequestModelConverterUnitTest.java:6]: 2023

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java (hunk 1 above)
Comment (FiveOFive) [reply addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java]: Done

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java (hunk 2 above)
Comment (FiveOFive) [reply addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverterUnitTest.java]: Done

File: addOns/openapi/CHANGELOG.md (hunk 4 above)
Comment (FiveOFive) [reply addOns/openapi/CHANGELOG.md]: Moved under ### Fixed and updated message + ended with a dot.

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/RequestModelConverterUnitTest.java (hunk 5 above)
Comment (FiveOFive) [reply addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/converter/swagger/RequestModelConverterUnitTest.java]: Updated

**Comment by FiveOFive** [comment]
I've made updates based on the comments. Thanks for the quick review.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**